File: packages/language-identification/android/src/main/java/io/capawesome/capacitorjs/plugins/mlkit/languageidentification/classes/results/IdentifyPossibleLanguagesResult.java
package io.capawesome.capacitorjs.plugins.mlkit.languageidentification.classes.results;

import androidx.annotation.NonNull;
import com.getcapacitor.JSArray;
import com.getcapacitor.JSObject;
import com.google.mlkit.nl.languageid.IdentifiedLanguage;
import io.capawesome.capacitorjs.plugins.mlkit.languageidentification.interfaces.Result;
import java.util.List;

public class IdentifyPossibleLanguagesResult implements Result {

    @NonNull
    private final List<IdentifiedLanguage> identifiedLanguages;

    public IdentifyPossibleLanguagesResult(@NonNull List<IdentifiedLanguage> identifiedLanguages) {
        this.identifiedLanguages = identifiedLanguages;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSArray identifiedLanguagesArray = new JSArray();
        for (IdentifiedLanguage identifiedLanguage : identifiedLanguages) {
            JSObject identifiedLanguageObject = new JSObject();
            identifiedLanguageObject.put("language", identifiedLanguage.getLanguageTag());
            identifiedLanguageObject.put("confidence", identifiedLanguage.getConfidence());
            identifiedLanguagesArray.put(identifiedLanguageObject);
        }
        JSObject result = new JSObject();
        result.put("identifiedLanguages", identifiedLanguagesArray);
        return result;
    }
}

File: packages/language-identification/ios/Plugin/Classes/Results/IdentifyPossibleLanguagesResult.swift
import Foundation
import Capacitor
import MLKitLanguageID

@objc public class IdentifyPossibleLanguagesResult: NSObject, Result {
    let identifiedLanguages: [IdentifiedLanguage]

    init(identifiedLanguages: [IdentifiedLanguage]) {
        self.identifiedLanguages = identifiedLanguages
    }

    @objc public func toJSObject() -> AnyObject {
        var identifiedLanguagesResult = JSArray()
        for identifiedLanguage in identifiedLanguages {
            var identifiedLanguageResult = JSObject()
            identifiedLanguageResult["language"] = identifiedLanguage.languageTag
            identifiedLanguageResult["confidence"] = identifiedLanguage.confidence
            identifiedLanguagesResult.append(identifiedLanguageResult)
        }
        var result = JSObject()
        result["identifiedLanguages"] = identifiedLanguagesResult
        return result as AnyObject
    }
}
